tests/utils/services.go

```diff
@@ -0,0 +1,47 @@
+package utils
+
+import (
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+func GetServiceInNamespaceOrDir(c *kubernetes.Clientset, ns, name string) *v1.Service {
+	svc, err := GetServiceInNamespace(c, ns, name)
+	if err != nil {
+		glog.Fatal(err)
+	}
+	return svc
+}
+
+func GetServiceInNamespace(c *kubernetes.Clientset, ns, name string) (*v1.Service, error) {
+	svc, err := c.CoreV1().Services(ns).Get(name, metav1.GetOptions{})
+	if err != nil {
```

Comment:
@awels  has incorporated retry logic into all/most of the funcs that call a k8s resource api. The goal being to reduce chances of a test failing just because the api was busy or non-responsive for a moment. k8s e2e does this for some api calls and not for others. Thoughts?